integration/kubernetes/init.sh

```diff
@@ -15,28 +15,61 @@ source "${SCRIPT_PATH}/../../.ci/lib.sh"
 source "${SCRIPT_PATH}/../../lib/common.bash"
 source "/etc/os-release" || source "/usr/lib/os-release"
 
+# Whether running on bare-metal mode or not.
+BAREMETAL="${BAREMETAL:-false}"
 CI=${CI:-""}
+# Path to the cluster network configuration file. Some architectures will
+# overwritten that variable.
+network_plugin_config="${network_plugin_config:-}"
 RUNTIME=${RUNTIME:-containerd-shim-kata-v2}
 RUNTIME_PATH=${RUNTIME_PATH:-$(command -v $RUNTIME)}
+CRI_RUNTIME="${CRI_RUNTIME:-crio}"
+
+untaint_node() {
+	# Enable the master node to be able to schedule pods.
+	local node_name="$(hostname)"
+	local get_taints="kubectl get 'node/${node_name}' -o jsonpath='{.spec.taints}'"
+	if eval $get_taints | grep -q 'NoSchedule'; then
+		info "Taint 'NoSchedule' is found. Untaint the node so pods can be scheduled."
+		kubectl taint nodes "${node_name}" \
+			node-role.kubernetes.io/master:NoSchedule-
+	fi
```

Comment:
Isn't tainting idempotent? I don't think the check is necessary.

Reply:
untainting isn't idempotent, it is going to fail with the taint doesn't exist. Unfortunately I messed with my environment so I cannot double-check at the moment.